### PR TITLE
Add validations for unknown and unsupported NamespaceSelector mode

### PR DESCRIPTION
**Pull request by nan-yu**

### CODE REVIEW

File: cmd/nomoserrors/examples/examples.go [hunk 1]
```diff
@@ -321,6 +321,12 @@ func Generate() AllExamples {
 	// 1069
 	result.add(validate.SelfReconcileError(fake.RootSyncV1Beta1(configsync.RootSyncName)))
 
+	// 1070
+	result.add(selectors.UnsupportedNamespaceSelectorModeError(fake.NamespaceSelector()))
+
+	// 1071
+	result.add(selectors.UnknownNamespaceSelectorModeError(fake.NamespaceSelector()))
```
Comment (janetkuo) [inline cmd/nomoserrors/examples/examples.go:328]: Is it necessary to separate unsupported vs. unknown? The error can just be "unsupported / invalid" to let users know that the specification is not valid.

Comment (nan-yu) [reply cmd/nomoserrors/examples/examples.go:328]: `unknown` applies to both hierarchy mode and unstructured mode, while `unsupported` only applies the hierarchy mode.

We have KNV1014 for invalidSelector, which applies to both ClusterSelector and NamespaceSelector: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/importer/analyzer/transform/selectors/errors.go#L29. There is also a TODO to refactor it properly. 

What's the best practice to reuse error code? Do we need to group them for a particular feature?

File: pkg/validate/tree/hydrate/namespace_selector_hydrator.go [hunk 2]
```diff
@@ -59,6 +59,13 @@ func labelSelector(obj ast.FileObject) (labels.Selector, status.Error) {
 	}
 	nss := s.(*v1.NamespaceSelector)
 
+	if nss.Spec.Mode == v1.NSSelectorDynamicMode {
+		return nil, selectors.UnsupportedNamespaceSelectorModeError(nss)
+	}
+	if nss.Spec.Mode != "" && nss.Spec.Mode != v1.NSSelectorStaticMode {
+		return nil, selectors.UnknownNamespaceSelectorModeError(nss)
+	}
```
Comment (janetkuo) [inline pkg/validate/tree/hydrate/namespace_selector_hydrator.go:67]: Can we just return `InvalidSelectorError(obj, err)` here (combined with below), and `err` tells users the cause, e.g. "dynamic mode isn't supported in hierachical repositories" etc.

File: cmd/nomoserrors/examples/examples.go (hunk 1 above)
Comment (janetkuo) [reply cmd/nomoserrors/examples/examples.go:328]: Please see my other comment

File: pkg/validate/tree/hydrate/namespace_selector_hydrator.go (hunk 2 above)
Comment (nan-yu) [reply pkg/validate/tree/hydrate/namespace_selector_hydrator.go:67]: Updated to reuse KNV1014, but created separate functions to wrap the error message, which seems the common way to build error message.

File: cmd/nomoserrors/examples/examples.go (hunk 1 above)
Comment (nan-yu) [reply cmd/nomoserrors/examples/examples.go]: changed to reuse KNV1014

**Review by janetkuo — COMMENTED** [review]
/lgtm 

#1031 should be merged first

**Review by janetkuo — APPROVED** [review]
/lgtm

**Pull request merged**